src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -110,32 +107,70 @@ internal async Task SearchAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task SearchCurrentDocumentAsync(CancellationToken cancellationToken)
+        {
+            if (_activeDocument == null)
+                return;
+
+            var project = _activeDocument.Project;
+            var service = _host.GetNavigateToSearchService(project);
+            if (service == null)
+                return;
+
+            await _progress.AddItemsAsync(1, cancellationToken).ConfigureAwait(false);
+            try
+            {
+                await service.SearchDocumentAsync(
+                    _activeDocument, _searchPattern, _kinds,
+                    r => _callback.AddItemAsync(project, r, cancellationToken),
+                    cancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                await _progress.ItemCompletedAsync(cancellationToken).ConfigureAwait(false);
+            }
+        }
+
         private async Task SearchAllProjectsAsync(bool isFullyLoaded, CancellationToken cancellationToken)
         {
+            var seenItems = new HashSet<INavigateToSearchResult>(NavigateToSearchResultComparer.Instance);
             var orderedProjects = GetOrderedProjectsToProcess();
-            var foundItems = await ProcessProjectsAsync(orderedProjects, isFullyLoaded, cancellationToken).ConfigureAwait(false);
+            var searchGeneratedDocuments = isFullyLoaded;
 
-            // If we're fully loaded then we're done at this point.  All the searches would have been against the latest
-            // computed data and we don't need to do anything else.
-            if (isFullyLoaded)
-                return;
+            var projectCount = orderedProjects.Sum(g => g.Length);
 
-            // We weren't fully loaded *but* we reported some items to the user, then consider that good enough for now.
-            // The user will have some results they can use, and (in the case that we actually examined the cache for
-            // data) we will tell the user that the results may be incomplete/inaccurate and they should try again soon.
-            if (foundItems)
-                return;
+            if (!isFullyLoaded)
```

Comment:
at the top level, once we decide to search all projects we do one of the following depending on if the solution is fully loaded or not.

1. If not fully loaded: First search all cached docs.  Then, if nothing was found, load everything and search all loaded docs.
2. If fully loaded: First search all normal loaded docs.  Then search generated docs.

Reply:
For 1), in the second step where we search the loaded docs when theres nothing cached, these are whatever projects we have partially loaded at that time right?  Since we're not fully loaded there may be projects we have no idea exist.  If so, it might be helpful to make a note of that